controllers/btpoperator_controller_test.go

```diff
@@ -872,3 +1194,97 @@ func checkIfNoBtpResourceExists() {
 func canIgnoreErr(err error) bool {
 	return k8serrors.IsNotFound(err) || meta.IsNoMatchError(err) || k8serrors.IsMethodNotSupported(err)
 }
+
+func replaceCaBundleInValidatingWebhooks(i []byte) bool {
+	webhookConfig := &admissionregistrationv1.ValidatingWebhookConfigurationList{}
+	err := k8sClient.List(ctx, webhookConfig, managedByLabelFilter)
+	Expect(err).To(BeNil())
+	Expect(len(webhookConfig.Items) > 0).To(BeTrue())
+	if len(webhookConfig.Items) > 0 {
+		webhook := webhookConfig.Items[0]
+		if len(webhook.Webhooks) > 0 {
+			webhook.Webhooks[0].ClientConfig.CABundle = i
+			err := k8sClient.Update(ctx, &webhook)
+			Expect(err).To(BeNil())
+			//time.Sleep(time.Second * 10)
+			return true
+		}
+	}
+	return false
+}
+
+func replaceCaBundleInMutatingWebhooks(i []byte) bool {
+	webhookConfig := &admissionregistrationv1.MutatingWebhookConfigurationList{}
+	err := k8sClient.List(ctx, webhookConfig, managedByLabelFilter)
+	Expect(err).To(BeNil())
+	Expect(len(webhookConfig.Items) > 0).To(BeTrue())
+	if len(webhookConfig.Items) > 0 {
+		webhook := webhookConfig.Items[0]
+		if len(webhook.Webhooks) > 0 {
+			webhook.Webhooks[0].ClientConfig.CABundle = i
+			err := k8sClient.Update(ctx, &webhook)
+			Expect(err).To(BeNil())
+			//time.Sleep(time.Second * 10)
+			return true
+		}
+	}
+	return false
+}
+
+func replaceSecretData(secret *corev1.Secret, key string, value []byte, key2 string, value2 []byte) {
+	data := secret.Data
+	if key != "" && value != nil {
+		data[key] = value
+	}
+
+	if key2 != "" && value2 != nil {
+		data[key2] = value2
+	}
+	secret.Data = data
+	err := k8sClient.Update(ctx, secret)
+	Expect(err).To(BeNil())
+}
+
+func getSecret(name string) *corev1.Secret {
+	secret := &corev1.Secret{}
+	err := k8sClient.Get(ctx, client.ObjectKey{Namespace: ChartNamespace, Name: name}, secret)
+	Expect(err).To(BeNil())
+	return secret
+}
+
+func checkHowManySecondsToExpiration(name string) float64 {
+	data, err := reconciler.getDataFromSecret(ctx, name)
+	Expect(err).To(BeNil())
+	key, err := reconciler.mapSecretNameToSecretDataKey(name)
+	Expect(err).To(BeNil())
+	value, err := reconciler.getValueByKey(reconciler.buildKeyNameWithExtension(key, CertificatePostfix), data)
+	Expect(err).To(BeNil())
+	decoded, _ := pem.Decode(value)
+	cert, err := x509.ParseCertificate(decoded.Bytes)
+	Expect(err).To(BeNil())
+	diff := cert.NotAfter.Sub(time.Now())
+	return diff.Seconds()
+}
+
+func ensureAllWebhooksManagedByBtpOperatorHaveCorrectCABundles() {
+	secret := getSecret(CaSecret)
+	ca, ok := secret.Data[reconciler.buildKeyNameWithExtension(CaSecretDataPrefix, CertificatePostfix)]
+	Expect(ok).To(BeTrue())
+	Expect(ca).To(Not(BeNil()))
+	vw := &admissionregistrationv1.ValidatingWebhookConfigurationList{}
+	err := k8sClient.List(ctx, vw, managedByLabelFilter)
+	Expect(err).To(BeNil())
+	for _, w := range vw.Items {
```

Comment:
Loops could be extracted as a separate function and used for both kinds of webhooks.

Reply:
I see it now. Thanks.